Custom Router: delegates invokes from NN if custom Router is provided (#5500)
mapbox/mapbox-navigation-android: CHANGELOG.md

Mapbox welcomes participation and contributions from everyone.

## Unreleased

> - :warning: Deprecated `RouterOrigin.Custom(obj: Any?)`. SDK is not supported holding object in Custom Route Origin, it becomes `null` when an object returns in Routers observer. [#5500](https://github.com/mapbox/mapbox-navigation-android/pull/5500)

Reviewer: Maybe?
```suggestion
- :warning: Deprecated `RouterOrigin.Custom(obj: Any?)`. SDK doesn't save `RouterOrigin.Custom#obj`. The `obj` filed will be `null` when the object is returned in a Routers observer. [#5500](https://github.com/mapbox/mapbox-navigation-android/pull/5500)
```

Author: changed to 
```suggestion
- :warning: Deprecated `RouterOrigin.Custom(obj: Any?)`. The SDK doesn't keep `RouteOrigin.Custom#obj` anymore, it always becomes `null` when the `obj` returns in `RoutesObserver`. [#5500](https://github.com/mapbox/mapbox-navigation-android/pull/5500)
```
